src/controllers/auth.controller.js

```diff
@@ -0,0 +1,263 @@
+
+  if (userData) {
+    throw ApiError.badRequest('Only for not authorized');
+  }
+
+  if (!name || !email || !password) {
+    throw ApiError.badRequest('Not all credentials are provided!');
+  }
+
+  const user = await userService.getByEmail(email);
+
+  if (user) {
+    throw ApiError.badRequest('The user with this email is already exists!');
+  }
+
+  const hashedPassword = await bcrypt.hash(password, +process.env.HASH_ROUNDS);
+  const newUser = { name, email, password: hashedPassword };
+
+  emailService.sendVerificationEmail(email, activationToken);
+  userService.create({ ...newUser, activationToken });
+
+  res.sendStatus(201);
+};
+
+const sendAuthentication = async (res, user) => {
+  const normalizedUser = userService.normalize(user);
+  const refreshToken = jwtService.generateRefreshToken(normalizedUser);
+  const accessToken = jwtService.generateAccessToken(normalizedUser);
+
+  await tokenService.save(normalizedUser.id, refreshToken);
+
+  await res.cookie('refreshToken', refreshToken, {
+    maxAge: 30 * 24 * 60 * 60 * 1000,
+    httpOnly: true,
+    sameSite: 'none',
+    secure: true,
+  });
+
+  res.send({ user: normalizedUser, accessToken });
+};
+
+const activate = async (req, res) => {
+  const { token } = req.params;
+  const { refreshToken } = req.cookies;
+
+  if (refreshToken) {
+    throw ApiError.badRequest('Only for not authorized');
+  }
+
+  const user = await User.findOne({ where: { activationToken: token } });
+
+  if (!user) {
+    throw ApiError.notFound();
+  }
+
+  user.activationToken = null;
+  user.save();
+
+  const normalizedUser = userService.normalize(user);
+  // const accesToken = jwtService.generateAccessToken(normalizedUser);
+
+  await sendAuthentication(res, normalizedUser);
+};
+
+const login = async (req, res) => {
+  const { email, password } = req.body;
+  const { refreshToken } = req.cookies;
+  const userData = await jwtService.validateRefreshToken(refreshToken);
+
+  if (userData) {
+    throw ApiError.badRequest('Only for not authorized');
+  }
+
+  if (!email || !password) {
+    throw ApiError.badRequest('Not all the credentials are provided!');
+  }
+
+  const user = await userService.getByEmail(email);
+
+  if (!user) {
+    throw ApiError.notFound();
+  }
+
+  const isPasswordValid = await bcrypt.compare(password, user.password);
+
+  if (!isPasswordValid) {
+    throw ApiError.badRequest('The username and/or password is incorrect.');
+  }
+
+  if (user.activationToken) {
+    throw ApiError.badRequest('The user`s email is not verified!');
+  }
+
+  sendAuthentication(res, user);
+};
+
+const refresh = async (req, res) => {
+  const refreshToken = req.cookies.refreshToken || '';
+  const token = await tokenService.getByToken(refreshToken);
+
+  if (!refreshToken || !token) {
+    throw ApiError.unauthorized();
+  }
+
+  const userData = jwtService.validateRefreshToken(refreshToken);
+
+  if (!userData) {
+    throw ApiError.badRequest('Invalid token!');
```

Comment:
The error message 'Invalid token!' might be misleading if the token is expired or revoked. Consider providing a more descriptive error message.